app/qml/PreviewPanel.qml

```diff
@@ -58,6 +59,15 @@ Item {
                         anchors.left: titleText.right
                         anchors.right: parent.right
 
+                        MouseArea {
+                            anchors.fill: parent
+                            propagateComposedEvents: true
+                            onClicked: {
+                                previewPanel.editClicked()
+                                mouse.accepted = false
+                            }
+                        }
```

Comment:
I would prefer to avoid the tricks with propagateComposedEvents... In fact when clicked on edit icon, we only want editClicked() to be triggered, not both editClicked() and contentClicked(). Then in onEditClicked() handler we can simply add a line with `stateManager.state = "form"` to switch to full-size window. This gives us also more flexibility for the future and the code would be easier to understand.